core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -96,6 +96,9 @@
     @Nullable
     private String strVal;
 
+    @Nullable
+    private volatile List<Consumer<? super ClientRequestContext>> customizers;
```

Comment:
Can we just make this final and not nullable by returning an empty list?

Reply:
We clean it up after retrieving the list in `runThreaedLocalContextCustomizers` (just in case there's any big garbage lurking in there), so we can't make it final.

I thought about using an empty list instead of null, but I chose to use null because it's performance-critical path. I can use an empty list if you mind, though.

Reply:
Thak makes sense. It's fine as it is. 😄

Reply:
We may as well document it, I think it's something like `// It is more common to have no customizers than to have any. So instead of defaulting to an empty list and removing null checks, we use null checks which are faster than checking if a list is empty.`

Reply:
Done